src/components/editor.rs

```diff
@@ -2906,104 +2906,115 @@ pub(crate) enum DispatchEditor {
-                FilterPush(_) => "FilterPush",
-                FilterClear => "FilterClear",
-                CursorAddToAllSelections => "CursorAddToAllSelections",
-                CyclePrimarySelection(_) => "CyclePrimarySelection",
-                CursorKeepPrimaryOnly => "CursorKeepPrimaryOnly",
-                ReplacePattern { config: _ } => "ReplacePattern",
-                Undo => "Undo",
-                Redo => "Redo",
-                KillLine(_) => "KillLine",
-                DeleteWordBackward { short: _ } => "DeleteWordBackward",
-                ReplaceCurrentSelectionWith(_) => "ReplaceCurrentSelectionWith",
-                TryReplaceCurrentLongWord(_) => "TryReplaceCurrentLongWord",
-                SelectLineAt(_) => "SelectLineAt",
-                ShowKeymapLegendNormalMode => "ShowKeymapLegendNormalMode",
-                ShowKeymapLegendInsertMode => "ShowKeymapLegendInsertMode",
-                Paste {
-                    direction: _,
-                    use_system_clipboard: _,
-                } => "Paste",
-                SwapCursorWithAnchor => "SwapCursorWithAnchor",
-                MoveCharacterBack => "MoveCharacterBack",
-                MoveCharacterForward => "MoveCharacterForward",
-                ShowKeymapLegendHelp => "ShowKeymapLegendHelp",
-                DeleteSurround(_) => "DeleteSurround",
-                ChangeSurround { from: _, to: _ } => "ChangeSurround",
-                Replace(_) => "Replace",
-                ApplyPositionalEdits(_) => "ApplyPositionalEdits",
-                ReplaceWithPreviousCopiedText => "ReplaceWithPreviousCopiedText",
-                ReplaceWithNextCopiedText => "ReplaceWithNextCopiedText",
-                MoveToLastChar => "MoveToLastChar",
-                PipeToShell { command: _ } => "PipeToShell",
-                ShowCurrentTreeSitterNodeSexp => "ShowCurrentTreeSitterNodeSexp",
-                Indent => "Indent",
-                Dedent => "Dedent",
-                #[cfg(test)]
-                _ => "",
+        match self {
+            Surround(open, close) => write!(f, "Surround {open}{close}",),
+            ShowJumps { .. } => write!(f, "{}", "Show Jumps"),
+            ScrollPageDown => write!(f, "{}", "Scroll Page Down"),
+            ScrollPageUp => write!(f, "{}", "Scroll Page Up"),
+            Transform(transformation) => write!(f, "{}", transformation),
+            SetSelectionMode(_, selection_mode) => {
+                write!(
+                    f,
+                    "Set Selection Mode - {}",
+                    convert_case::Casing::to_case(
+                        &format!("{:?}", selection_mode),
+                        convert_case::Case::Title
+                    )
+                )
             }
-        )
+            MoveSelection(movement) => write!(f, "Move Selection - {movement:?}",),
+            Save => write!(f, "{}", "Save"),
+
+            // TODO: show details for the following
```

Comment:
@yves-biener it's not done yet, would you like to continue?

If you wish to continue you should merge this PR into your branch and continue from your branch